### PR TITLE
simdjson: add 0.8.2 + fix compiler versions verifications

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **simdjson/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Raise the minimum compile versions? or revert the code?

**Comment by SpaceIm** [comment]
The problem is that gcc 7 docker images in CCI are based on gcc 7.2, while simdjson seems to require gcc 7.4 at least.

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by a4z — APPROVED** [review]

File: recipes/simdjson/all/conanfile.py [hunk 1]
```diff
@@ -32,8 +38,8 @@ def _build_subfolder(self):
     @property
     def _compilers_minimum_version(self):
         return {
-            "gcc": "7.4",
-            "Visual Studio": "15.7",
+            "gcc": "8",
+            "Visual Studio": "16",
```
Comment (mathbunnyru) [inline recipes/simdjson/all/conanfile.py:42]: The author tells that gcc 7 is enough.
Isn't that right?
And clang 6 should work as well.

https://github.com/simdjson/simdjson#quick-start

Comment (mathbunnyru) [reply recipes/simdjson/all/conanfile.py:42]: Oh, I just saw the comment about 7.4, is it possible to keep 7.4

Comment (SpaceIm) [reply recipes/simdjson/all/conanfile.py:42]: No, due to CI of CCI.

Comment (prince-chrismc) [reply recipes/simdjson/all/conanfile.py:42]: Well if we revert the changes... than will pass for CCI and anyone using custom compiler versions like it was before

**Review by uilianries — APPROVED** [review]

File: recipes/simdjson/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/simdjson/all/conanfile.py:42]: I don't know if it's very common, there is no silver bullet here.

**Pull request merged**